homeassistant/components/recorder/core.py

```diff
@@ -689,14 +696,14 @@ def _process_one_task_or_recover(self, task: RecorderTask) -> None:
         # happens to rollback and recover
         self._reopen_event_session()
 
-    def _setup_recorder(self) -> None | int:
-        """Create connect to the database and get the schema version."""
+    def _setup_recorder(self) -> bool:
```

Comment:
This is called in `_handle_sqlite_corruption` as well but I'm not sure there is anything we can do at that point if its `False`

Reply:
Yes, I saw that. In `_handle_sqlite_corruption` we didn't check the return value, so I don't think it matters that we now return a success flag instead of the schema version?